src/Raven.Server/Monitoring/Snmp/SnmpWatcher.cs

```diff
@@ -382,8 +392,16 @@ private async Task AddDatabases()
                     {
                         context.CloseTransaction();
 
-                        var result = await _server.ServerStore.SendToLeaderAsync(new UpdateSnmpDatabasesMappingCommand(missingDatabases, RaftIdGenerator.NewId()));
-                        await _server.ServerStore.Cluster.WaitForIndexNotification(result.Index);
+                        try
+                        {
+                            var result = await _server.ServerStore.SendToLeaderAsync(new UpdateSnmpDatabasesMappingCommand(missingDatabases, RaftIdGenerator.NewId()));
+                            await _server.ServerStore.Cluster.WaitForIndexNotification(result.Index);
+                        }
+                        catch (Exception e) when (e is TimeoutException || e is OperationCanceledException)
+                        {
+                            // we will do the update in AddDatabaseIfNecessary
```

Comment:
Which is hooked to OnDatabaseLoaded event, so we'll do it on the actual database load, right?

Reply:
Yes. Updated the comment